include/unifex/linux/io_uring_context.hpp

```diff
@@ -441,7 +441,8 @@ class io_uring_context::read_sender {
 
     void start_io() noexcept {
       UNIFEX_ASSERT(context_.is_running_on_io_thread());
-
+      stopCallback_.construct(
+            get_stop_token(receiver_), cancel_callback{*this});
```

Comment:
In the case that `start()` is invoked on a "remote" thread, you're deferring stop callback construction until after we've thunked to the IO thread.  Is this intentional?  Why not construct the stop callback in `start()`?

Also, what happens if the stop token has already received a stop request by the time you construct the stop callback?  For `inplace_stop_token`, the callback constructor will invoke the callback before returning—do you handle that scenario?